src/DotLiquid/Context.cs

```diff
@@ -28,10 +28,20 @@ public class Context
 
         private readonly ErrorsOutputMode _errorsOutputMode;
 
+        private SyntaxCompatibility _syntaxCompatibilityLevel;
+
         /// <summary>
         /// Liquid syntax flag used for backward compatibility
         /// </summary>
-        public SyntaxCompatibility SyntaxCompatibilityLevel { get; set; }
+        public SyntaxCompatibility SyntaxCompatibilityLevel {
+            get => _syntaxCompatibilityLevel;
+            set
+            {
+                if (_strainer != null)
+                    throw new System.ArgumentException("");

```

Comment:
This needs a proper exception to disallow changing the compatibility level after strainers are computed.